fix: disable injectable-pipe migration

crisbeto opened the pull request:
Disables the injectable pipe migration until we can decide whether this is the right solution for Ivy. Rolling it out properly will involve a more detailed plan and more changes like updating the styleguide, scaffolding schematics etc.

Context for the new `test-migrations.json`: since we use the `migrations.json` both for the real migrations and for tests, it doesn't allow us to disable a schematic while continuing to run its tests. This change adds the test-specific file so that we can continue running the `injectable-pipe` tests, even though the schematic itself is disabled.

AndrewKushnir commented:
[Presubmit](http://test/OCL:245655361:BASE:245655367:1556480590671:32c1fd8a)

IgorMinar reviewed (approved):
Lgtm but can you please extend the commit message to contain more details. The problem is that we are not yet sure if this is the right solution for ivy and if it is we need to do much more to roll out this change: e.g. update style guide, scaffolding schematics, docs and tutorial, etc etc. So it's better to wait until we are sure that we want this change and then make a comprehensive plan on how to roll it out.

IgorMinar commented:
@crisbeto Thanks for putting this together so quickly.

crisbeto committed:
fix: disable injectable-pipe migration

Disables the injectable pipe migration until we can decide whether this is the right solution for Ivy. Rolling it out properly will involve a more detailed plan and more changes like updating the styleguide, scaffolding schematics etc.

Context for the new `test-migrations.json`: since we use the `migrations.json` both for the real migrations and for tests, it doesn't allow us to disable a schematic, but continue running its tests. This change adds the test-specific file so that we can continue running the `injectable-pipe` tests, even though the schematic itself is disabled.

crisbeto commented:
Updated the commit message and PR description with more context.

devversion reviewed (approved):
Ideally there would have been only a new collection for the `injectable-pipe` schematic in order to run tests without shipping it. Using the real migration collection allows us to validate that the schematics (which are intended to run publicly) are wired up correctly.

I don't mind too much though. LGTM :smile:

AndrewKushnir commented:
[Presubmit #2](http://test/OCL:245655361:BASE:245662578:1556489630400:d3d26306) (after additional changes)

crisbeto commented:
@devversion I can see it going either way. I did it this way, because it's more consistent.

devversion commented:
@crisbeto True it's more consistent for the tests, but given that it means that we need to duplicate schematic collection entries and we no longer test the publicly enabled migrations with the actual `ng-update` collection, I think we should change it back and only have a collection for the injectable-pipe. 

Doesn't have to be part of this PR as it already had various presubmits..
